src/rdkafka_broker.c

```diff
@@ -768,6 +768,21 @@ void rd_kafka_broker_conn_closed(rd_kafka_broker_t *rkb,
         rd_kafka_broker_fail(rkb, log_level, err, "%s", errstr);
 }
 
+rd_bool_t buf_contains_toppar(rd_kafka_buf_t *rkbuf, rd_kafka_toppar_t *rktp) {
+
+        if (rd_list_cnt(&rkbuf->rkbuf_u.Produce.batch_list) > 0) {
+                /* this is multi-batch request, loop through all batches */
+                rd_kafka_msgbatch_t *msgbatch;
+                int i;
+                RD_LIST_FOREACH(msgbatch, &rkbuf->rkbuf_u.Produce.batch_list, i) {
```

Comment:
What's the potential size of this list?

Reply:
The list contains all of "ready" batches in this iteration of Processing. The size of list is bounded by number of topic+partition this broker thread manages, and it's likely to be significantly smaller than that, given not all toppars has ready batch (linger, batch-size constraints) in each iteration.  I will add line of comments